library/cartridge_instance.py

```diff
@@ -14,21 +15,115 @@
 }
 
 
+def serialize_dict(d):
+    parts = []
+
+    for k, v in d.items():
+        value_str = None
+        if type(v) == str:
+            value_str = "'%s'" % v
+        elif type(v) == float or type(v) == int:
+            value_str = "%s" % v
+        elif type(v) == bool:
+            value_str = 'true' if v else 'false'
+        else:
+            raise CartridgeException(
+                cartridge_errcodes.BAD_VALUE_TYPE,
+                "Unknown value type: {}" % type(v)
+            )
+
+        parts.append("%s = %s" % (k, value_str))
+
+    return '{ %s }' % ', '.join(parts)
+
+
+def box_cfg_was_called(control_console):
+    return control_console.eval('''
+        return type(box.cfg) ~= 'function'
+    ''')
+
+
+def get_box_cfg(control_console):
+    return control_console.eval('''
+        return box.cfg or box.NULL
+    ''')
+
+
+def change_memory_size(param_name, cartridge_defaults, config, control_console):
+    new_memory_size = None
+    if param_name in config:
+        new_memory_size = config[param_name]
+    else:
+        new_memory_size = cartridge_defaults[param_name]
+
+    if new_memory_size is None:
+        return False
+
+    # Get current memory size
+    current_memory_size = control_console.eval('''
+        return type(box.cfg) ~= 'function' and box.cfg.{} or box.NULL
+    '''.format(param_name))
+
+    if current_memory_size is None:
+        # box.cfg wasn't called
+        return False
+
+    if new_memory_size <= current_memory_size:
+        return False
+
+    # try to increase memory size
+    increased = control_console.eval('''
+        local ok, err = pcall(function()
+            box.cfg {{ {param_name} = {new_memory_size} }}
+        end)
+        if not ok then
+            if tostring(err):find("cannot decrease memory size at runtime") == nil then
+                error('failed to set {param_name}: ' .. tostring(err))
+            end
+        end
+        return ok
+    '''.format(param_name=param_name, new_memory_size=new_memory_size))
+
+    return increased
+
+
+def change_dynamic_params(cartridge_defaults, config, control_console):
+    params = dict()
+
+    # Get current values
+    current_box_cfg = get_box_cfg(control_console)
+    if current_box_cfg is None:
+        return False
+
+    for param_name, param_value in config.items():
+        if param_name in dynamic_box_cfg_params and param_name not in memory_size_box_cfg_params:
```

Comment:
I think you need some one-line helpers here to improve readability: `is_dynamic_parameter`/`is_memory_management_parameter` to eliminate strange chains of conditions